### PR TITLE
osd: don't get inc osdmap name truncated

### PR DESCRIPTION
**Pull request by xiexingguo**

### CODE REVIEW

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -6642,7 +6642,7 @@ void OSD::handle_osd_map(MOSDMap *m)
       OSDMap *o = new OSDMap;
       if (e > 1) {
 	bufferlist obl;
-	get_map_bl(e - 1, obl);
+	assert(get_map_bl(e - 1, obl));
```
Comment (liewegas) [inline src/osd/OSD.cc]: please assert on a different line.. we don't want any side-effects in the assert itself

File: src/osd/OSD.h [hunk 2]
```diff
@@ -1146,7 +1146,7 @@ class OSD : public Dispatcher,
     return ghobject_t(hobject_t(sobject_t(object_t(foo), 0)));
   }
   static ghobject_t get_inc_osdmap_pobject_name(epoch_t epoch) {
-    char foo[20];
+    char foo[32];
```
Comment (tchaikov) [inline src/osd/OSD.h]: @xiexingguo so it should be 22, right?

Comment (xiexingguo) [inline src/osd/OSD.h]: @tchaikov You are definitely right. Thanks.

**Comment by tchaikov** [comment]
lgtm

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2016-05-13_16:00:42-rados-wip-yuri-testing-distro-basic-smithi/

**Pull request merged**